controllers/cruisecontroloperation_controller.go

```diff
@@ -0,0 +1,473 @@
+		return requeueAfter(defaultRequeueIntervalInSeconds)
+	}
+
+	// Filtering out CruiseControlOperation by kafka cluster ref and state
+	var ccOperationsKafkaClusterFiltered []*banzaiv1alpha1.CruiseControlOperation
+	for i := range ccOperationListClusterWide.Items {
+		operation := &ccOperationListClusterWide.Items[i]
+		ref, err := getKafkaClusterReference(operation)
+		if err != nil {
+			log.Info(err.Error())
+		}
+		if ref.Name == kafkaClusterRef.Name && ref.Namespace == kafkaClusterRef.Namespace &&
+			operation.IsCurrentTaskOperationValid() && !operation.IsDone() {
+			ccOperationsKafkaClusterFiltered = append(ccOperationsKafkaClusterFiltered, operation)
+		}
+	}
+
+	// Update currentTask states from Cruise Control
+	err = r.updateCurrentTasks(ctx, ccOperationsKafkaClusterFiltered)
+	if err != nil {
+		log.Error(err, "requeue event as updating state of currentTask(s) failed")
+		return requeueAfter(defaultRequeueIntervalInSeconds)
+	}
+	//When the task is not in execution we can remove the finalizer
+	if isFinalizeNeeded(currentCCOperation) && !currentCCOperation.IsCurrentTaskRunning() {
+		controllerutil.RemoveFinalizer(currentCCOperation, cruiseControlOperationFinalizer)
+		if err := r.Update(ctx, currentCCOperation); err != nil {
+			return requeueWithError(log, err.Error(), err)
+		}
+		return reconciled()
+	}
+
+	// Sorting operations into categories which are sorted by priority
+	ccOperationQueueMap := sortOperations(ccOperationsKafkaClusterFiltered)
+	// When there is no more job present in the cluster we reconciled.
+	if len(ccOperationQueueMap[ccOperationForFinalize]) == 0 && len(ccOperationQueueMap[ccOperationFirstExecution]) == 0 &&
+		len(ccOperationQueueMap[ccOperationRetryExecution]) == 0 && len(ccOperationQueueMap[ccOperationInProgress]) == 0 {
+		log.Info("there is no more operation for execution")
+		return reconciled()
+	}
+
+	ccOperationExecution := selectOperationForExecution(ccOperationQueueMap)
+	// There is nothing to be executed for now, requeue
+	if ccOperationExecution == nil {
+		return requeueAfter(defaultRequeueIntervalInSeconds)
+	}
+
+	// Check if CruiseControl is ready as we cannot perform any operation until it is in ready state unless it is a stop execution operation
+	if (status.InExecution() || len(ccOperationQueueMap[ccOperationInProgress]) > 0) && ccOperationExecution.GetCurrentTaskOp() != banzaiv1alpha1.OperationStopExecution {
+		// Requeue becuse we can't do more
+		return requeueAfter(defaultRequeueIntervalInSeconds)
+	}
+
+	log.Info("executing Cruise Control task", "operation", ccOperationExecution.GetCurrentTaskOp(), "parameters", ccOperationExecution.GetCurrentTaskParameters())
+	// Executing operation
+	cruseControlTaskResult, err := r.executeOperation(ccOperationExecution)
+
+	if err != nil {
+		log.Error(err, "Cruise Control task execution got an error", "operation", ccOperationExecution.GetCurrentTaskOp(), "parameters", ccOperationExecution.GetCurrentTaskParameters())
+	}
```

Comment:
Question: why is it enough to just log the error? Shouldn't we perform other actions when it fails to execute the cruise control operation?

Reply:
thanks for the explanation, can you add this as a comment along with the code? Just to make things more clear about why logging the error is enough here